app/controllers/dfe_sign_in_controller.rb

```diff
@@ -12,14 +16,19 @@ def callback
 private
 
   def default_authenticated_path
-    if authorized_for_support_interface?
+    if get_support_user
       support_interface_path
     else
       provider_interface_path
     end
   end
 
-  def authorized_for_support_interface?
-    SupportUser.load_from_session(session)
+  def get_support_user
+    if @already_tried # using separate var as guard as lookup can return nil, which is then cached
```

Comment:
I think you can use `defined?(@user)` if you want to avoid the extra variable here.

Reply:
Thanks, went for modified version of Tijmen's one liner.